katharsis-client/src/test/java/io/katharsis/client/ClientTest.java

```diff
@@ -0,0 +1,172 @@
+package io.katharsis.client;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import io.katharsis.client.mock.models.Project;
+import io.katharsis.client.mock.models.Task;
+import io.katharsis.queryParams.QueryParams;
+
+public class ClientTest extends AbstractClientTest {
+
+	@Test
+	public void testFindEmpty() {
+		List<Task> tasks = taskRepo.findAll(new QueryParams());
+		Assert.assertTrue(tasks.isEmpty());
+	}
+
+	@Test
+	public void testFindNull() {
+		try {
+			taskRepo.findOne(1L, new QueryParams());
+			Assert.fail();
+		} catch (ClientException e) {
+			Assert.assertEquals("Not Found", e.getMessage());
+		}
+	}
+
+	@Test
+	public void testSaveAndFind() {
+		Task task = new Task();
+		task.setId(1L);
+		task.setName("test");
+		taskRepo.save(task);
+
+		// check retrievable with findAll
+		List<Task> tasks = taskRepo.findAll(new QueryParams());
+		Assert.assertEquals(1, tasks.size());
+		Task savedTask = tasks.get(0);
+		Assert.assertEquals(task.getId(), savedTask.getId());
+		Assert.assertEquals(task.getName(), savedTask.getName());
+
+		// check retrievable with findAll(ids)
+		tasks = taskRepo.findAll(Arrays.asList(1L), new QueryParams());
+		Assert.assertEquals(1, tasks.size());
+		savedTask = tasks.get(0);
+		Assert.assertEquals(task.getId(), savedTask.getId());
+		Assert.assertEquals(task.getName(), savedTask.getName());
+
+		// check retrievable with findOne
+		savedTask = taskRepo.findOne(1L, new QueryParams());
+		Assert.assertEquals(task.getId(), savedTask.getId());
+		Assert.assertEquals(task.getName(), savedTask.getName());
+	}
+
+	@Test
+	public void testGeneratedId() {
+		Task task = new Task();
+		task.setId(null);
+		task.setName("test");
+		Task savedTask = taskRepo.save(task);
+		Assert.assertNotNull(savedTask.getId());
+	}
+
+	@Test
+	public void testDelete() {
+		Task task = new Task();
+		task.setId(1L);
+		task.setName("test");
+		taskRepo.save(task);
+
+		taskRepo.delete(1L);
+
+		List<Task> tasks = taskRepo.findAll(new QueryParams());
+		Assert.assertEquals(0, tasks.size());
+	}
+
+	@Test
+	@Ignore // not supported by spec
+	public void testSaveOneRelation() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Fix or remove this skipped unit test [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1607)